ReactWindows/ReactNative.Shared/Bridge/Queue/MessageQueueThread.cs

```diff
@@ -173,7 +189,7 @@ protected override void Enqueue(Action action)
 
             protected override bool IsOnThreadCore()
             {
-                return CoreWindow.GetForCurrentThread().Dispatcher != null;
+                return SynchronizationContext.Current != null;
             }
```

Comment:
How does this confirm that we're on the dispatcher thread?  This seems to only confirms that there is a current synchronization context...